README.md

```diff
@@ -21,67 +17,226 @@ Additionally you also need to mark some of the dependencies in your dependencies
+### Further config options
+
+| Option                   | Default                        | CLI or project.clj    | Description | Example |
+|--------------------------|--------------------------------|-----------------------|-------------|---------|
+| project-prefix           | mranderson{mranderson version} | CLI                   | project pecific prefix to use when shadowing            | `lein source-deps :project-prefix cider.inlined-deps` |
+| skip-javaclass-repackage | false                          | CLI                   | If true [Jar Jar Links](https://code.google.com/p/jarjar/) won't be used to repackage java classes in dependencies            | `lein source-deps :skip-javaclass-repackage true`        |
+| prefix-exclusions        | empty list                     | CLI                   | List of prefixes which should not be processed in imports            |  `lein source-deps :prefix-exclusions "[\"classlojure\"]"`  |
+| watermark                | :mranderson/inlined            | project.clj           | When processing namespaces in dependencies MrAnderson marks them with a meta so inlined namespaces can be identified. Helpful for tools like [cljdoc](https://cljdoc.org)            | `:mranderson {:watermark nil}` to switch off watermarking or provide your own keyword        |
+| shadowing-only           | false                          | CLI, project.clj      | Switch between **unresolved tree** and **shadowing only** mode | `lein source-deps :shadowing-only true` |
+| overrides                | empty list                     | project.clj           | Defines dependency overrides in **unresolved tree** mode | `:mranderson {:overrides {[mvxcvi/puget fipp] [fipp "0.6.15"]}}` |
+| expositions              | empty list                     | project.clj           | Makes transient dependencies available in the project's source files in **unresolved tree** mode | `:mranderson {:expositions [[mvxcvi/puget fipp]]}` |
+
+## Prerequisites
+
+Leiningen 2.8.3 or above. For MrAnderson to work, does not mean your project is restricted to a java or clojure version.
+
+## Is it good? Should I use it?
+
+Yes and yes. Use it if you have depedency conflicts and don't care to solve them. Or if you don't want your library's dependencies to interfere with the dependencies of your users' (leiningen plugins is a good example here). Or if you want to explore a bit more in the hellish land of dependency handling.
+
+### Projects that use MrAnderson
+
+- [cider-nrepl](https://github.com/clojure-emacs/cider-nrepl)
+- [refactor-nrepl](https://github.com/clojure-emacs/refactor-nrepl)
+- [re-frame-10x](https://github.com/Day8/re-frame-10x)
+
+### Libraries that work with MrAnderson
```

Comment:
What's the purpose of this section? I guess by default most libs would work, right?

Reply:
most would I suppose. But there are corner cases (and bugs of course) One example I've come across with when inlining deps for mranderson itself is the library, claypoole I first used for parallelisation. The problematic bit is here: this import prefixes two classes of claypoole itself. One of those classes is created with a deftype, the other one is created by a java class. MrAnderson prefixing deftype based classes in imports is a separate step from prefixing java class based classes in imports. The logic prefixing them is also different resulting in different prefixes. This means that inlining this library breaks now because the deftype prefixing "wins" for this import but this would render the java class based import wrong.

Now mranderson could be improved to handle this of course. Either aline the two steps so they generate the same prefix or split this import into two so they can have different prefixes.

The point is here however that there are quite a few corner cases in this space (import, requires, dependencies between nses) so when focusing on a release for example I just rather pick my fights. Therefore for a while it would make sense to maintain a white list and black list kind of thing.